README.md

```diff
@@ -102,6 +102,32 @@ snapshots_url = build_log.match(%r{https://eigen-ci.s3.amazonaws.com/\d+/index.h
 fail("There were [snapshot errors](#{snapshots_url})") if snapshots_url
 ```
 
+## Plugins
+
+Danger was built with a platform in mind: It can be used with any kind of software project and allows you to write your own action to have structured source code.
+
+In your `Dangerfile` you can import local or remote actions using
+
+```ruby
+import "./danger_plugins/work_in_progress_warning"
+import "https://raw.githubusercontent.com/danger/danger/master/danger_plugins/work_in_progress_warning.rb"
+
+# Call those actions using
+work_in_progress_warning
+
+custom_plugin(variable: "value")
+```
+
+To create a new plugin run
+
+```
+danger new_plugin
+```
+
+This will generate a new Ruby file which you can modify to fit your needs. 
+
+If you think your new plugin is useful for other developers too, feel free to submit a PR to danger by copying the action to `lib/danger/plugins`. 
```

Comment:
> We could have one 'gem', `danger-useful`

Terminology wise, I think `danger-core` would work here. If it's essentials, it should ship with danger. There's no point in over-abstracting too far yet.

The reason I call this a blocker is that I can see how much time & effort it takes felix to maintain fastlane, and that's got a big audience. Danger could have an even bigger audience, given it's usefulness in most dev communities. So, we want to make it _really, really_ easy for others to build on top of danger, and not _into_ danger. Specifically, so that we're not spending time on minutae of actions instead of building a better platform.

If it makes sense to use something in every danger file ( like `protect_files` does ) then it should be in with Danger, if we feel like it would be better for all of our provided actions to go into `danger-core` cool, but it needs to be all or nothing, and do we need that extra level of abstraction yet?